site/docs/references/mixin-syntax.md

```diff
@@ -1,124 +1,253 @@
 
-Every CSS class defined in a **stylable** CSS file can be used as a mixin.
+```css
+/* CSS mixin file - mixins.st.css */
+.root {
+    color: purple;
+}
 
-``` css
+.someClass {
+    color: green;
+}
+```
 
-/* file a.st.css */
-.a{
-    color:red;
+``` css
+/* CSS file - example.st.css - imports the above mixin */
+:import {
+    -st-from: './mixins.st.css';
+    -st-default: MyComp;
+    -st-named: someClass;
 }
 
-/* file b.st.css */
-:import{
-    -st-from:'./a.st.css';
-    -st-named:a;
+.rootMixedIn {
+    -st-mixin: MyComp; /* stylesheet mixin */
 }
-.b{
-    -st-mixin:a;
+
+.classMixedIn {
+    -st-mixin: someClass; /* class mixin */
 }
 ```
 
-``` css
+```css
 /* CSS output */
-.b{
-    color:red;
+.rootMixedIn {
+    color: purple; /* from stylesheet mixin */
 }
-```
 
-## Defining mixins in JavaScript or TypeScript
+.rootMixedIn .someClass { /* ruleset added as a result of the stylesheet mixin */
+    color: green;
+}
 
-You can easily create new mixins using JavaScript or TypeScript.
+.classMixedIn {
+    color: green; /* from class mixin */
+}
+```
 
-## Applying multiple mixins
+### CSS mixin with parameters and variables
 
-You can apply multiple mixins in the same line.
+CSS mixins can accept named parameters in the following format:
+ `mixin(variableName valueOverride)`. 
+ 
+ Using parameters in a mixin enables you to override specific [variables](./variables.md) inside of a mixin before they are applied.
 
+Here is an example of using a variable in a CSS mixin and how it can be overriden by the mixin's parameter value.
 
 ```css
 /* CSS */
-.c {
-    /* apply multiple mixins */
-    -st-mixin: noParams, multiParams(param1, param2);
+:vars {
+    color1: green;
 }
 
-```
+.classToMixin {
+    background: value(color1);
+}
 
-## Usage with special characters
+.targetClass {
+    -st-mixin: classToMixin(color1 orange);
+}
+```
 
 ```css
-/* CSS */
-.c { /* use quotations to include comma */
-    -st-mixin: mix(300, "xx,x"); /* ["300", "xx,x"] */
+/* CSS output */
+.classToMixin {
+    background: green; /* from local class */
 }
-.d { /* escape slashes */
-    -st-mixin: mix(300, "\"xxx\""); /* ["300", "\"xxx\""] */
+
+.targetClass {
+    background: orange; /* from mixin with override */
 }
 ```
 
-## How mixins are applied
+## JavaScript mixins
 
-Mixins can add CSS declarations to the CSS ruleset to which they are applied:
+JavaScript mixins allow you to create complex structures in CSS based on the arguments passed to the mixin and the mixin logic. 
 
-* Rules are added at the position in the CSS where the `-st-mixin` is declared.
-* Any selectors that are appended as a result of the mixin are added directly after the ruleset that the mixin was applied to.
-* Multiple mixins are applied according to the order that they are specified.
+A JS mixin returns a CSS fragment which can be a single declaration and its value, multiple declarations, multiple rulesets, or any combination of these.
```

Comment:
"single declaration and its value" - declaration is a `prop:value` definition, "and its value" is not needed.
I would shorten this into:

A JS mixin returns a CSS fragment which can contain multiple declarations with optional sub style-rules